packages/contracts-bedrock/tasks/genesis-l1.ts

```diff
@@ -83,20 +77,26 @@ task('genesis-l1', 'create a genesis config')
         berlinBlock: 0,
         londonBlock: 0,
         clique: {
-          period: 15,
+          period: deployConfig.l1BlockTime,
           epoch: 30000,
         },
       },
-      nonce: '0x0',
-      timestamp: ethers.BigNumber.from(timestamp).toHexString(),
-      extraData:
-        '0x0000000000000000000000000000000000000000000000000000000000000000ca062b0fd91172d89bcd4bb084ac4e21972cc4670000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
-      gasLimit: ethers.BigNumber.from(15_000_000).toHexString(),
-      difficulty: '0x1',
-      mixHash:
-        '0x0000000000000000000000000000000000000000000000000000000000000000',
-      coinbase: '0x0000000000000000000000000000000000000000',
+      nonce: deployConfig.l1GenesisBlockNonce,
+      timestamp: ethers.BigNumber.from(deployConfig.l1GenesisBlockTimestamp).toHexString(),
+      extraData: ethers.utils.hexConcat([
+        ethers.constants.HashZero,
+        deployConfig.cliqueSignerAddress,
+        ethers.utils.hexZeroPad('0x', 65),
+      ]),
+      gasLimit: deployConfig.l1GenesisBlockGasLimit,
+      difficulty: deployConfig.l1GenesisBlockDifficulty,
+      mixHash: deployConfig.l1GenesisBlockMixHash,
+      coinbase: deployConfig.l1GenesisBlockCoinbase,
       alloc,
+      number: deployConfig.l1GenesisBlockNumber,
+      gasUsed: deployConfig.l1GenesisBlockGasUsed,
+      parentHash: deployConfig.l1GenesisBlockParentHash,
```

Comment:
Parent hash should always always be `ethers.constants.HashZero` right?

Reply:
Not if we start doing migration tests with previous chain history. But generally, yes.